docs/usage/self-hosted-configuration.md

```diff
@@ -128,7 +128,16 @@ e.g.
 
 ## composerIgnorePlatformReqs
 
-Set to `false` to prevent usage of `--ignore-platform-reqs` in the Composer package manager.s
+By default, Renovate will run Composer with `--ignore-platform-reqs` as the PHP platform used by Renovate most probably won't match with the required PHP environment of your project as configured in your `composer.json` file.
+However, this also means that all platform constraints (including PHP version) will be ignored by default, which can result in updated dependencies that are not compatible with your platform.
+
+To solve this, you should configure explicit ignored platform requirements (for example `ext-zip`) by setting them separately in this array.
+Each item will be added to the Composer command with `--ignore-platform-req`, resulting in it being ignored during its invocation.
+
+If an empty array is configured, Renovate uses its default behaviour.
+
+Set to `null` to fully omit `--ignore-platform-reqs/--ignore-platform-req` during Composer invocation.
```

Comment:
@rarkins I think we want to get rid of using `null`? And instead more descriptive terms like `auto/disabled/enabled`? 

- https://github.com/renovatebot/renovate/issues/9100

Do we want to replace the current `null` with `auto` maybe?

Reply:
In my opinion `auto` here should mean that renovate automatically extracts all platform requirements from the project configuration to add them to the ignore requirements (except for PHP itself), but that would be hard to accomplish as not all requirements are available just from the repository (dependencies can also configure certain requirements). See also https://github.com/renovatebot/renovate/issues/2355#issuecomment-849431921.

Secondly, I do not believe combining textual options with a String option is currently supported by the configuration definition: https://github.com/renovatebot/renovate/blob/main/lib/config/types.ts#L350.

Reply:
@HonkingGoose This is a string array option now. so #9100 did not apply here.

Reply:
Oh okay. 😄 Sorry for the confusion!